src/main/java/org/broadinstitute/hellbender/tools/walkers/cancer/contamination/ContaminationEstimate.java

```diff
@@ -0,0 +1,181 @@
+            precalculatedEpsilon[i] = Math.pow(10.0,-1.0*(((double)i)/10.0));
+        }
+    }
+
+    /**
+     * create the contamination estimate, given:
+     * @param precision the precision value, to what level are we calculating the contamination
+     */
+    public ContaminationEstimate(double precision,
+                                 double maf,
+                                 byte[] bases,
+                                 byte[] quals,
+                                 byte arrayAllele,
+                                 byte hapmapAlt,
+                                 String popName,
+                                 Locatable locus
+    ) {
+        // setup the bins to the correct precision
+        this.precision = precision;
+        bins = new double[(int)Math.ceil(100/precision)+1];
+        if (maf == 0) maf = 0.00001;
+
+        popultationName = popName;
+
+        Arrays.fill(bins,0.0); // just to make sure we don't have any residual values
+
+        // convert the quals
+        double[] realQuals = new double[quals.length];
+        int qIndex = 0;
+        for (byte qual : quals) {realQuals[qIndex++] = Math.pow(10.0,-1.0*(qual/10.0));}
+
+        // check our inputs
+        if (maf > 1.0 || maf < 0.0) throw new IllegalArgumentException("Invalid allele Freq: must be between 0 and 1 (inclusive), maf was " + maf + " for population " + popName);
+
+        // calculate the contamination for each bin
+        int qualOffset = 0;
+        for (byte base : bases) {
+
+            if (base == arrayAllele) { arrayAlleleObservations++; }
+            if (base == hapmapAlt) { alternateAlleleObservations++; }
+            double epsilon = precalculatedEpsilon[quals[qualOffset++]];
+
+            for (int index = 0; index < bins.length; index++) {
+
+
+                double contaminationRate = (1.0 - (double) index / (double) bins.length);
+
+                if (base == arrayAllele) {
+                    bins[index] += Math.log((1.0 - contaminationRate) * (1.0 - epsilon) +
+                            contaminationRate * ((maf) * (1.0 - epsilon) + (1.0 - maf) * (epsilon/3.0)));
+                    populationFit += Math.log(epsilon);
+
+                } else if(hapmapAlt == base) {
+                    bins[index] += Math.log((1.0 - contaminationRate) * (epsilon / 3.0) +
+                            contaminationRate * ((maf) * (epsilon/3.0) + (1.0 - maf) * (1.0 - epsilon)));
+
+                    populationFit += Math.log(maf + epsilon);
+                }
+            }
+        }
+    }
+
+    public double[] getBins() {
+        return bins;
+    }
+
+    public void setPopulationFit(double populationFit) {
+        this.populationFit = populationFit;
+    }
+
+    public double getPopulationFit() {
+        return populationFit;
+    }
+
+    public String getPopultationName() {
+        return popultationName;
+    }
+
+    public static class ConfidenceInterval {
```

Comment:
Can this be replaced with Sam's posterior distribution stuff from the ACNV MCMC code?